File: app/src/main/java/com/weatherforecast/android/ui/weatherChat/WeatherViewModel.java
package com.weatherforecast.android.ui.weatherChat;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

public class WeatherViewModel extends AndroidViewModel {

    public WeatherViewModel(Application application) {
        super(application);
    }

    public LiveData<String> getWeatherInfo() {
        MutableLiveData<String> weatherData = new MutableLiveData<>();
        // 假设获取天气信息的逻辑
        weatherData.setValue("今天的天气是晴天，温度 25°C，适宜外出。");
        return weatherData;
    }
}
package com.weatherforecast.android.ui.weather;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

public class WeatherViewModel extends AndroidViewModel {

    public WeatherViewModel(Application application) {
        super(application);
    }

    public LiveData<String> getWeatherInfo() {
        MutableLiveData<String> weatherData = new MutableLiveData<>();
        // 假设获取天气信息的逻辑
        weatherData.setValue("今天的天气是晴天，温度 25°C，适宜外出。");
        return weatherData;
    }
}

File: app/src/main/java/com/weatherforecast/android/ui/weather/weathershow/WeatherViewModel.kt
package com.weatherforecast.android.ui.weather.weathershow

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.switchMap
import com.weatherforecast.android.logic.Repository
import com.weatherforecast.android.logic.model.weather.Location

class WeatherViewModel : ViewModel(){
    private val locationLiveData = MutableLiveData<Location>()

    var locationLng = ""

    var locationLat = ""

    var placeName = ""

    var placeAddress = ""

    var placeRealtimeTem = -100

    var placeSkycon = ""

    var isUpdatePlaceManage = false

    val weatherLiveData = locationLiveData.switchMap { location ->
        Repository.refreshWeather(location.lng,location.lat)
    }

    fun refreshWeather(lng: String, lat: String) {
        locationLiveData.value =
            Location(lng, lat)
    }
}